qa/tasks/cephfs/test_admin.py

```diff
@@ -219,14 +219,45 @@ def is_required(index):
 
             if index % 3 == 0:
                 continue;
-            self.fs.mon_manager.raw_cluster_cmd('fs', 'required_client_features', self.fs.name, 'add', feature)
+            self.fs.required_client_features('add', feature)
             self.assertTrue(is_required(index))
 
             if index % 2 == 0:
                 continue;
-            self.fs.mon_manager.raw_cluster_cmd('fs', 'required_client_features', self.fs.name, 'rm', feature)
+            self.fs.required_client_features('rm', feature)
             self.assertFalse(is_required(index))
 
+    def test_required_client_feature_add_reserved(self):
+        """
+        That `ceph fs required_client_features X add reserved` fails.
+        """
+
+        try:
+            p = self.fs.required_client_features('add', 'reserved')
+        except CommandFailedError:
+            self.assertIn('Invalid feature name', p.stderr.getvalue())
```

Comment:
Variable `p` won't be available here IIUC.

Reply:
```
2021-01-07 13:20:38,241.241 INFO:__main__:======================================================================
2021-01-07 13:20:38,241.241 INFO:__main__:ERROR: test_required_client_feature_add_reserved (tasks.cephfs.test_admin.TestAdminCommands)
2021-01-07 13:20:38,241.241 INFO:__main__:That `ceph fs required_client_features X add reserved` fails.
2021-01-07 13:20:38,241.241 INFO:__main__:----------------------------------------------------------------------
2021-01-07 13:20:38,241.241 INFO:__main__:Traceback (most recent call last):
2021-01-07 13:20:38,241.241 INFO:__main__:  File "/home/rishabh/repos/ceph/master/qa/tasks/cephfs/test_admin.py", line 236, in test_required_client_feature_add_reserved
2021-01-07 13:20:38,242.242 INFO:__main__:    p = self.fs.required_client_features('add', 'reserved')
2021-01-07 13:20:38,242.242 INFO:__main__:  File "/home/rishabh/repos/ceph/master/qa/tasks/cephfs/filesystem.py", line 584, in required_client_features
2021-01-07 13:20:38,242.242 INFO:__main__:    return self.mon_manager.run_cluster_cmd(args=c, stdout=StringIO(), stderr=StringIO())
2021-01-07 13:20:38,242.242 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 995, in run_cluster_cmd
2021-01-07 13:20:38,242.242 INFO:__main__:    return self.controller.run(**kwargs)
2021-01-07 13:20:38,242.242 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 399, in run
2021-01-07 13:20:38,242.242 INFO:__main__:    return self._do_run(**kwargs)
2021-01-07 13:20:38,242.242 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 467, in _do_run
2021-01-07 13:20:38,242.242 INFO:__main__:    proc.wait()
2021-01-07 13:20:38,242.242 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 213, in wait
2021-01-07 13:20:38,243.243 INFO:__main__:    raise CommandFailedError(self.args, self.exitstatus)
2021-01-07 13:20:38,243.243 INFO:__main__:teuthology.exceptions.CommandFailedError: Command failed with status 22: ['./bin/ceph', 'fs', 'required_client_features', 'cephfs', 'add', 'reserved']
2021-01-07 13:20:38,243.243 INFO:__main__:
2021-01-07 13:20:38,243.243 INFO:__main__:During handling of the above exception, another exception occurred:
2021-01-07 13:20:38,243.243 INFO:__main__:
2021-01-07 13:20:38,243.243 INFO:__main__:Traceback (most recent call last):
2021-01-07 13:20:38,243.243 INFO:__main__:  File "/home/rishabh/repos/ceph/master/qa/tasks/cephfs/test_admin.py", line 238, in test_required_client_feature_add_reserved
2021-01-07 13:20:38,243.243 INFO:__main__:    self.assertIn('Invalid feature name', p.stderr.getvalue())
2021-01-07 13:20:38,243.243 INFO:__main__:UnboundLocalError: local variable 'p' referenced before assignment
```

Reply:
```
2021-01-07 13:20:38,241.241 INFO:__main__:======================================================================
2021-01-07 13:20:38,241.241 INFO:__main__:ERROR: test_required_client_feature_add_reserved (tasks.cephfs.test_admin.TestAdminCommands)
2021-01-07 13:20:38,241.241 INFO:__main__:That `ceph fs required_client_features X add reserved` fails.
2021-01-07 13:20:38,241.241 INFO:__main__:----------------------------------------------------------------------
2021-01-07 13:20:38,241.241 INFO:__main__:Traceback (most recent call last):
2021-01-07 13:20:38,241.241 INFO:__main__:  File "/home/rishabh/repos/ceph/master/qa/tasks/cephfs/test_admin.py", line 236, in test_required_client_feature_add_reserved
2021-01-07 13:20:38,242.242 INFO:__main__:    p = self.fs.required_client_features('add', 'reserved')
2021-01-07 13:20:38,242.242 INFO:__main__:  File "/home/rishabh/repos/ceph/master/qa/tasks/cephfs/filesystem.py", line 584, in required_client_features
2021-01-07 13:20:38,242.242 INFO:__main__:    return self.mon_manager.run_cluster_cmd(args=c, stdout=StringIO(), stderr=StringIO())
2021-01-07 13:20:38,242.242 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 995, in run_cluster_cmd
2021-01-07 13:20:38,242.242 INFO:__main__:    return self.controller.run(**kwargs)
2021-01-07 13:20:38,242.242 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 399, in run
2021-01-07 13:20:38,242.242 INFO:__main__:    return self._do_run(**kwargs)
2021-01-07 13:20:38,242.242 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 467, in _do_run
2021-01-07 13:20:38,242.242 INFO:__main__:    proc.wait()
2021-01-07 13:20:38,242.242 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 213, in wait
2021-01-07 13:20:38,243.243 INFO:__main__:    raise CommandFailedError(self.args, self.exitstatus)
2021-01-07 13:20:38,243.243 INFO:__main__:teuthology.exceptions.CommandFailedError: Command failed with status 22: ['./bin/ceph', 'fs', 'required_client_features', 'cephfs', 'add', 'reserved']
2021-01-07 13:20:38,243.243 INFO:__main__:
2021-01-07 13:20:38,243.243 INFO:__main__:During handling of the above exception, another exception occurred:
2021-01-07 13:20:38,243.243 INFO:__main__:
2021-01-07 13:20:38,243.243 INFO:__main__:Traceback (most recent call last):
2021-01-07 13:20:38,243.243 INFO:__main__:  File "/home/rishabh/repos/ceph/master/qa/tasks/cephfs/test_admin.py", line 238, in test_required_client_feature_add_reserved
2021-01-07 13:20:38,243.243 INFO:__main__:    self.assertIn('Invalid feature name', p.stderr.getvalue())
2021-01-07 13:20:38,243.243 INFO:__main__:UnboundLocalError: local variable 'p' referenced before assignment
```

Reply:
hits the same error for me
```UnboundLocalError: local variable 'p' referenced before assignment```

Reply:
Yes, the issue lies with the code.

Reply:
thanks, fixed